plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1817,6 +1821,8 @@ bool producer_plugin_impl::remove_expired_trxs( const fc::time_point& deadline )
             ++num_expired;
    });
 
+   _metrics->unapplied_transactions.value = orig_count - num_expired;
```

Comment:
I think it is a good plan to only update this at the start of each block instead of every time the queue is modified. I would move all these metric updates into a method and then call that method from `start_block()`.

Also, this could just be `_metrics->unapplied_transactions.value = _unapplied_transactions.size();`